### PR TITLE
fix: add path type in acceptedFiles

### PR DESCRIPTION
**Pull request by aslan-alt**
**What kind of change does this PR introduce?**
- [x] Bug Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Style
- [ ] Build
- [ ] Chore
- [ ] Documentation
- [ ] CI

**Did you add tests for your changes?**
- [ ] Yes, my code is well tested
- [x] Not relevant

**If relevant, did you update the documentation?**
- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**
fix ts error

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by rolandjitsu** [comment]
@aslan-alt we cannot do this because the type of the files depends on what the `getFilesFromEvent` returns. So we use `T extends File` in places where we return file objects. I think you can set the `T` to the type you expect.

File: typings/react-dropzone.d.ts [hunk 1]
```diff
@@ -66,8 +68,8 @@ export type DropzoneState = DropzoneRef & {
   isDragAccept: boolean;
   isDragReject: boolean;
   isFileDialogActive: boolean;
-  draggedFiles: File[];
-  acceptedFiles: File[];
+  draggedFiles: FileWithPath[];
+  acceptedFiles: FileWithPath[];
```
Comment (rolandjitsu) [inline typings/react-dropzone.d.ts:72]: I've explained why you cannot do this. But we could use generics the same way we do for `onDrop`/`onDropAccepted`.

**Review by rolandjitsu — CHANGES_REQUESTED** [review]

File: typings/react-dropzone.d.ts (hunk 1 above)
Comment (aslan-alt) [reply typings/react-dropzone.d.ts]: The variable T cannot be used because here is a concrete value and not a function
I changed the way of writing, hope you like it

File: typings/tests/hook.tsx [hunk 2]
```diff
@@ -2,14 +2,6 @@ import React from "react";
 import { useDropzone, DropzoneProps } from "../../";
 
 export const Dropzone = ({ children, ...opts }: DropzoneProps) => {
-  const { ...state } = useDropzone(opts);
-  return children(state);
+  const state = useDropzone(opts);
+  return <>{children(state)}</>;
```
Comment (aslan-alt) [inline typings/tests/hook.tsx:6]: Import React from "react" is used here; if there is no <> there will be the following error
 @typescript-eslint/no-unused-vars

File: typings/react-dropzone.d.ts (hunk 1 above)
Comment (aslan-alt) [reply typings/react-dropzone.d.ts]: Screen Shot 2022-04-25 at 14 37 59

File: typings/tests/hook.tsx [hunk 3]
```diff
@@ -2,14 +2,6 @@ import React from "react";
...
 };
-
-<Dropzone>
-  {({ getRootProps, getInputProps }) => (
-    <div {...getRootProps()}>
-      <input {...getInputProps()} />
-    </div>
-  )}
-</Dropzone>;
```
Comment (rolandjitsu) [inline typings/tests/hook.tsx:15]: Why was this removed?

File: typings/tests/hook.tsx (hunk 2 above)
Comment (rolandjitsu) [reply typings/tests/hook.tsx:6]: I haven't seen that error so far. How did you get it?

File: typings/tests/hook.tsx (hunk 3 above)
Comment (gustavo-cardoso-deal) [reply typings/tests/hook.tsx:15]: I'm curious too

Comment (aslan-alt) [reply typings/tests/hook.tsx:15]: because this code is useless

Comment (aslan-alt) [reply typings/tests/hook.tsx:15]: I don't think this code makes any sense
because there are no exports and references

Comment (aslan-alt) [reply typings/tests/hook.tsx:15]: I don't think we've ever used these codes
because there are no exports and import
Screen Shot 2022-04-27 at 11 05 45

File: typings/tests/hook.tsx (hunk 2 above)
Comment (aslan-alt) [reply typings/tests/hook.tsx:6]: because I removed this useless code 🤣
Screen Shot 2022-04-27 at 11 20 04

File: typings/tests/hook.tsx (hunk 3 above)
Comment (rolandjitsu) [reply typings/tests/hook.tsx:15]: The purpose of it was to check that using the hook syntax doesn't cause any TS errors. So please add it back.

File: typings/tests/hook.tsx (hunk 2 above)
Comment (rolandjitsu) [reply typings/tests/hook.tsx:6]: Yeah, that code needs to be added back, then you won't get that warning.

Comment (aslan-alt) [reply typings/tests/hook.tsx:6]: This is an optimization, I can delete it if you don't like it

File: typings/tests/hook.tsx (hunk 3 above)
Comment (aslan-alt) [reply typings/tests/hook.tsx:15]: This is an optimization, I can delete it if you don't like it

Comment (aslan-alt) [reply typings/tests/hook.tsx]: ok, i have reverted the code
But I still want to know why not use the following way:
`
 return <>{children(state)}</>;
`
In this way, ts will not report errors, and the syntax is more streamlined

File: typings/tests/hook.tsx (hunk 2 above)
Comment (aslan-alt) [reply typings/tests/hook.tsx]: ok, i have reverted the code

File: typings/react-dropzone.d.ts [hunk 4]
```diff
@@ -66,8 +66,8 @@ export type DropzoneState = DropzoneRef & {
   isDragAccept: boolean;
   isDragReject: boolean;
   isFileDialogActive: boolean;
-  draggedFiles: File[];
-  acceptedFiles: File[];
+  draggedFiles: (File & { path?: string })[];
+  acceptedFiles: (File & { path?: string })[];
```
Comment (rolandjitsu) [inline typings/react-dropzone.d.ts:70]: Isn't `FileWithPath` the same thing?
```suggestion
  draggedFiles: FileWithPath[];
  acceptedFiles: FileWithPath[];
```
Could you try that and see if you still get errors?

Comment (aslan-alt) [reply typings/react-dropzone.d.ts:70]: I always see this error, see [this demo](https://github.com/aslan-alt/react-dropzone-ts-demo)

Comment (rolandjitsu) [reply typings/react-dropzone.d.ts:70]: @aslan-alt could you update to v14 and see if the same issue happens? I'll take a look later this week.

Comment (rolandjitsu) [reply typings/react-dropzone.d.ts:70]: I just tried it out and if you set:
```
 acceptedFiles: FileWithPath[];
```
as suggested, the error is gone.

So please update that and also set the `FileRejection.file` prop type to the same type.

Comment (aslan-alt) [reply typings/react-dropzone.d.ts]: okay

**Comment by GiancarlosIO** [comment]
👀

**Comment by aguscha333** [comment]
Any update on when this will be merged?

**Comment by Crayon-ShinChan** [comment]
Will this PR be merged? I noticed that this project hasn't accepted any new PRs for the past two years. guys, what's going on?

**Review by rolandjitsu — APPROVED** [review]
